julia/src/model.jl

```diff
@@ -72,31 +73,24 @@ mutable struct StanModel
 
         windows_dll_path_setup()
 
-        lib = Libc.Libdl.dlopen(lib)
+        lib = dlopen(lib)
 
         err = Ref{Cstring}()
-
-        stanmodel = ccall(
-            Libc.Libdl.dlsym(lib, "bs_model_construct"),
-            Ptr{StanModelStruct},
-            (Cstring, UInt32, Ref{Cstring}),
-            data,
-            seed,
-            err,
-        )
+        stanmodel = @ccall $(dlsym(lib, :bs_model_construct))(
```

Comment:
What does `@ccall ...` do to error messages, when wrongly typed arguments are passed? I mean, this question should not stop this change from going in, but I've had bad experiences with error messages coming from within a function call which lives inside a macro.

Reply:
Are you asking if an end user messes up the types or if we do? I think our functions would stop them before they hit this, but maybe not if we ourselves made a mistake?

Reply:
Ya, sorry, that wasn't clear.  I had in mind something like a user passing in a wrongly typed variable into `param_constrain()` or `log_density()` or something, but I see now / remember that all the functions I had in mind have typed parameters themselves.  So any type errors will catch there, and not here.